src/test/compile-middleware_test.ts

```diff
@@ -108,5 +108,60 @@ suite('compile-middleware', () => {
       // The valid script tag should still be compiled.
       assert.notInclude(response.text, `<script>\nclass A {}\n</script>`);
     });
+
+    suite('with compile option set to \'auto\'', () => {
+
+      beforeEach(() => {
+        app = getApp({
+          root: root,
+          compile: 'auto',
+          componentDir: path.join(root, 'bower_components'),
+        });
+        // Ensure a fresh cache for each test.
+        babelCompileCache.reset();
+      });
+
+      const userAgentsThatSupportES2015 = [
+        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.98 Safari/537.36',
```

Comment:
@aomarks added some UA strings and whether they should trigger compilation in browserNeedsCompilation. Maybe reuse those now?